kong/runloop/handler.lua

```diff
@@ -1513,7 +1513,21 @@ return {
       -- Report HTTP status for health checks
       local balancer_data = ctx.balancer_data
       if balancer_data and balancer_data.balancer_handle then
+        -- https://github.com/nginx/nginx/blob/2485681308bd8d3108da31546cb91bb97813a3fb/src/http/ngx_http_upstream.c#L5656
+        -- because of the way of nginx do the upstream_status variable, it can be
+        -- a string or a number, so we need to parse it to get the status
         local status = ngx.var.upstream_status
+        if tonumber(status) == nil then
```

Comment:
Here we have a simple solution to get the last upstream status code,
using `sub(ngx.var.upstream_status, -3)`.

There is a example code in `pdk/service/response.lua`.

Reply:
good idea